src/Common/src/Interop/Unix/libssl/SecuritySafeHandles.cs

```diff
@@ -249,11 +284,9 @@ public SafeDeleteContext(SafeFreeCredentials credential, bool isServer, bool rem
             }
             catch(Exception ex)
             {
-                if (gotCredRef)
-                {
-                    _credential.DangerousRelease();
-                }
                 Debug.Write("Exception Caught. - " + ex);
+                Debug.Assert((null != credential), "Null credential in SafeDeleteContext");
+                Dispose();
```

Comment:
@shrutigarg we need to (implement) and call base.Dispose which should also set _credential to null to prevent double-free